python/GafferSceneUI/ShaderTweaksUI.py

```diff
@@ -327,6 +328,26 @@ def __addTweak( self, name, plugTypeOrValue ) :
 		with Gaffer.UndoScope( self.getPlug().ancestor( Gaffer.ScriptNode ) ) :
 			self.getPlug().addChild( plug )
 
+class _ShaderTweakPlugValueWidget( GafferUI.TweakPlugValueWidget ) :
+
+	def __init__( self, plugs ):
+		GafferUI.TweakPlugValueWidget.__init__( self, plugs )
+		self._TweakPlugValueWidget__row.append(
+			GafferUI.MenuButton(
+				image="shaderTweakProxyIcon.png",
+				hasFrame=False,
+				menu=GafferUI.Menu( Gaffer.WeakMethod( self.__createProxyMenuDefinition ), title = "Create Proxy" ),
+				toolTip = "Proxies allow making connections from the outputs of nodes in the input network."
+			)
+		)
+
+	def __createProxyMenuDefinition( self ) :
+		return GafferSceneUI.ShaderTweakProxyUI._plugContextMenu( self, self.getPlug()["value"], self.getPlug().node() )
```

Comment:
This is passing `self` as the `graphEditor` argument to `_plugContextMenu()`. It seems like that doesn't make any sense, and I believe it may also be creating a circular reference that keeps this widget alive after its time. I've been getting a lot of these :

```
Internal C++ object (PySide2.QtWidgets.QWidget) already deleted.
```

Reply:
Passing the ScriptWindow around doesn't seem ideal, on a couple of counts. First, when we have floating NodeEditors or detached GraphEditors, the dialogue is parented to the main ScriptWindow and not the window in question - that might even be on another monitor. Second, doesn't baking the ScriptWindow into the menu commands all but guarantee a circular reference, since the menu will be parented to the ScriptWindow?

If this is just about getting the window to things like `__browseAffectedShaders()`, then there is an easier solution : add a `menu` argument to `__browseAffectedShaders()`. That will automatically be passed the menu, and you can then use `menu.ancestor( Window )` to find the appropriate window to parent a dialogue. That's the standard approach we take everywhere else.

Reply:
Addressed in 5c8784a